test/unit/org/apache/cassandra/tools/AuditLogViewerTest.java

```diff
@@ -61,6 +71,65 @@ public void tearDown() throws IOException
         }
     }
 
+    @Test
+    public void testNoArgs()
+    {
+        ToolRunner tool = runner.invokeTool(toolPath);
+        assertTrue(tool.getStdout(), tool.getStdout().contains("usage:"));
+        assertTrue(tool.getCleanedStderr(), tool.getCleanedStderr().contains("Audit log files directory path is a required argument."));
+        assertEquals(1, tool.getExitCode());
+    }
+
+    @Test
+    public void testMaybeChangeDocs()
+    {
+        // If you added, modified options or help, please update docs if necessary
+        ToolRunner tool = runner.invokeTool(toolPath, "-h");
+        assertEquals("34aea94756d4f6760b836dfbfa7bbf06", DigestUtils.md5Hex(tool.getStdout()));
+    }
+
+    @Test
+    public void testHelpArg()
+    {
+        Arrays.asList("-h", "--help").stream().forEach(arg -> {
+            ToolRunner tool = runner.invokeTool(toolPath, arg);
+            assertTrue(tool.getStdout(), tool.getStdout().contains("usage:"));
+            assertTrue(tool.getCleanedStderr(),tool.getCleanedStderr().isEmpty());
+            assertEquals(0, tool.getExitCode());
+        });
+    }
+
+    @Test
+    public void testIgnoreArg()
+    {
+        Arrays.asList("-i", "--ignore").stream().forEach(arg -> {
+            ToolRunner tool = runner.invokeTool(toolPath, path.toAbsolutePath().toString(), arg);
+            assertTrue(tool.getStdout(), tool.getStdout().isEmpty());
+            assertTrue(tool.getCleanedStderr(),
+                       tool.getCleanedStderr().isEmpty() // j8 is fine
+                       || tool.getCleanedStderr().startsWith("WARNING: An illegal reflective access operation has occurred")); //j11 throws an error
```

Comment:
Ok some explanation is needed here. On 15991 (this ticket) given the mammoth size of the main task (15583), I am only 'seeding' unit tests for all tools. These are focusing on testing the input args only. This has no other reason but to move forward at a pace I can handle and not bite off more than I can chew.

On [CASSANDRA-15583](https://issues.apache.org/jira/browse/CASSANDRA-15583#)'s description you'll see I raised individual tickets to UT each tool. CASSANDRA-16021 is the one for the `AuditLogViewer`. So this fix you're referring to belongs in that ticket. Same for any lack of coverage or any other bugs we find in any other of the tools.

Does it make sense? I'm just trying to break up the big task in manageable bits rather than fixing all problems on all the tools at once big-bang style.

Reply:
If you are planning to fix the jdk11 bug in a different ticket, can we move these tests to those tickets?  We can also leave it without the WARNING exclusion and mark the test as ignored (such as `org.apache.cassandra.distributed.test.RepairCoordinatorFailingMessageTest`).

Reply:
icwym +1-ish. Instead of ignoring the whole test, given it's useful to test the args even if only partially imo, why don't we 'ignore' the offending assert alone? I will comment it out with an explanation and add a comment to the other Jira to make sure this gets fixed there. wdyt?

Reply:
> why don't we 'ignore' the offending assert alone?

Just saw the change, so it looks like you comment out the exception; that is fine for now.